### PR TITLE
Emergency wipe option for the Skywallet

### LINKED ISSUE #2579: Create an emergency wipe option for the Skywallet
If the user enters several wrong PINs for the Skywallet, the device will increase the waiting time before allowing the user to try again. The problem is that in some cases the time may end being very long and the connection with the device may be lost during the process. If this happens, it is not possible to use the Skywallet option in the UI and it is not posible to wipe the device.

An option for wiping the device without the waiting time must be added.

### PR DESCRIPTION
**Pull request by Senyoret1**
Fixes #2579

Changes:
- A page for wiping the a Skywallet without having to perform for the normal procedure was added. In includes a warning for the user.

- A link for opening the new page when a inactivity error is found was added to the UI.

Does this change need to mentioned in CHANGELOG.md?
No

### CODE REVIEW

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { FormBuilder } from '@angular/forms';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+
+import { ChildHwDialogParams } from '../../layout/hardware-wallet/hw-options-dialog/hw-options-dialog.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
import { ChildHwDialogParams } from "../../layout/hardware-wallet/hw-options-dialog/hw-options-dialog.component";
```

File: src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts [hunk 2]
```diff
@@ -140,6 +150,15 @@ export class HwDialogBaseComponent<T> implements OnDestroy {
         result.link = AppConfig.hwWalletDaemonDownloadUrl;
       }
 
+      // If the operation was cancelled for inactivity, a link to open the page for wiping the
+      // Skywallet is added.
+      if (result.text === 'hardware-wallet.errors.timeout') {
+        result.text = 'hardware-wallet.errors.timeout-with-configurable-link';
+        result.linkText = 'force-skywallet-wipe.title';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts:157]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
        result.linkText = "force-skywallet-wipe.title";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
import { Router } from "@angular/router";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { Component } from '@angular/core';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
import { Component } from "@angular/core";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+export class ForceSkywalletWipeComponent {
+  constructor(
+    public formBuilder: FormBuilder,
+    private router: Router,
+    private dialog: MatDialog,
+  ) { }
+
+  proceed() {
+    const config = new MatDialogConfig();
+    config.width = '450px';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:27]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
    config.width = "450px";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { FormBuilder } from '@angular/forms';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:4]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
import { MatDialog, MatDialogConfig } from "@angular/material/dialog";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import { HwWipeDialogComponent } from '../../layout/hardware-wallet/hw-wipe-dialog/hw-wipe-dialog.component';
+
+/**
+ * Allows wipe a Skywallet. It is mainly for restoring the device when the user is not able to
+ * use it because the operations are always cancelled for inactivity.
+ */
+@Component({
+  selector: 'app-force-skywallet-wipe',
+  templateUrl: './force-skywallet-wipe.component.html',
+  styleUrls: ['./force-skywallet-wipe.component.scss'],
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:16]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
  styleUrls: ["./force-skywallet-wipe.component.scss"],
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { FormBuilder } from '@angular/forms';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+
+import { ChildHwDialogParams } from '../../layout/hardware-wallet/hw-options-dialog/hw-options-dialog.component';
+import { HwWipeDialogComponent } from '../../layout/hardware-wallet/hw-wipe-dialog/hw-wipe-dialog.component';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
import { HwWipeDialogComponent } from "../../layout/hardware-wallet/hw-wipe-dialog/hw-wipe-dialog.component";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { FormBuilder } from '@angular/forms';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
import { FormBuilder } from "@angular/forms";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+import { ChildHwDialogParams } from '../../layout/hardware-wallet/hw-options-dialog/hw-options-dialog.component';
+import { HwWipeDialogComponent } from '../../layout/hardware-wallet/hw-wipe-dialog/hw-wipe-dialog.component';
+
+/**
+ * Allows wipe a Skywallet. It is mainly for restoring the device when the user is not able to
+ * use it because the operations are always cancelled for inactivity.
+ */
+@Component({
+  selector: 'app-force-skywallet-wipe',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:14]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
  selector: "app-force-skywallet-wipe",
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    config.width = '450px';
+    config.autoFocus = false;
+
+    // Data for the modal window.
+    config.data = <ChildHwDialogParams> {
+      wallet: null,
+      requestOptionsComponentRefresh: ((error: string = null, recheckSecurityOnly: boolean = false) => {
+        if (!error) {
+          // Return to the wallet list after the operation is done.
+          this.router.navigate([''], {replaceUrl: true});
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:36]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
          this.router.navigate([""], {replaceUrl: true});
```

File: src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts [hunk 12]
```diff
@@ -140,6 +150,15 @@ export class HwDialogBaseComponent<T> implements OnDestroy {
         result.link = AppConfig.hwWalletDaemonDownloadUrl;
       }
 
+      // If the operation was cancelled for inactivity, a link to open the page for wiping the
+      // Skywallet is added.
+      if (result.text === 'hardware-wallet.errors.timeout') {
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts:155]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
      if (result.text === "hardware-wallet.errors.timeout") {
```

File: src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts [hunk 13]
```diff
@@ -140,6 +150,15 @@ export class HwDialogBaseComponent<T> implements OnDestroy {
...
+        result.text = 'hardware-wallet.errors.timeout-with-configurable-link';
+        result.linkText = 'force-skywallet-wipe.title';
+        result.link = '#/skywallet-wipe';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts:158]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
        result.link = "#/skywallet-wipe";
```

File: src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts [hunk 14]
```diff
@@ -140,6 +150,15 @@ export class HwDialogBaseComponent<T> implements OnDestroy {
         result.link = AppConfig.hwWalletDaemonDownloadUrl;
       }
 
+      // If the operation was cancelled for inactivity, a link to open the page for wiping the
+      // Skywallet is added.
+      if (result.text === 'hardware-wallet.errors.timeout') {
+        result.text = 'hardware-wallet.errors.timeout-with-configurable-link';
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/layout/hardware-wallet/hw-dialog-base.component.ts:156]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
        result.text = "hardware-wallet.errors.timeout-with-configurable-link";
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+        }
+      }),
+    };
+
+    this.dialog.open(HwWipeDialogComponent, config);
+  }
+
+  back() {
+    // Return to the wallet list.
+    this.router.navigate(['']);
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:46]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
    this.router.navigate([""]);
```

File: src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+import { ChildHwDialogParams } from '../../layout/hardware-wallet/hw-options-dialog/hw-options-dialog.component';
+import { HwWipeDialogComponent } from '../../layout/hardware-wallet/hw-wipe-dialog/hw-wipe-dialog.component';
+
+/**
+ * Allows wipe a Skywallet. It is mainly for restoring the device when the user is not able to
+ * use it because the operations are always cancelled for inactivity.
+ */
+@Component({
+  selector: 'app-force-skywallet-wipe',
+  templateUrl: './force-skywallet-wipe.component.html',
```
Comment (SkycoinSynth) [inline src/gui/static/src/app/components/pages/force-skywallet-wipe/force-skywallet-wipe.component.ts:15]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skycoin/pullRequest?prid=7163601)
```suggestion
  templateUrl: "./force-skywallet-wipe.component.html",
```

**Review by iketheadore — APPROVED** [review]

**Pull request merged**